source/program/remote_api.cpp

```diff
@@ -54,36 +76,83 @@ void PrepareThread() {
 
     int rval = nn::socket::Bind(g_TcpSocket, (struct sockaddr*)&serverAddr, sizeof(serverAddr));
     if (rval < 0) EXL_ABORT(69);
-    
+
     rval = nn::socket::Listen(g_TcpSocket, 1);
     if (rval < 0) EXL_ABORT(69);
 }
 
-void SocketSpawn(void*) {
+void AddPacketToSendBuffer(PacketBuffer buffer) {
+    nn::os::LockMutex(&sendBufferLock);
+    sendBuffer.push_back(buffer);
+    nn::os::UnlockMutex(&sendBufferLock);
+}
+
+void ReceiveLogic() {
+    // don't recv new packets while we waiting for game loop otherwise we need to make a copy of our receive buffer
+    if (readyForGameThread.load()) return;
+    memset(RecvBuffer.data(), 0, 4096);
```

Comment:
There should be a constant somewhere for this size. Probably `RecvBuffer.size()`.